Anlab.Mvc/Startup.cs

```diff
@@ -66,17 +71,69 @@ public void ConfigureServices(IServiceCollection services)
                 .AddEntityFrameworkStores<ApplicationDbContext>()
                 .AddDefaultTokenProviders();
 
+            //services.ConfigureApplicationCookie(options => options.LoginPath = "/Account/LogIn");
+
+            services.AddAuthentication()
+                .AddOpenIdConnect("UCDavis", options =>
+                {
+                    options.Events.OnRedirectToIdentityProvider = context =>
+                    {
+                        context.ProtocolMessage.SetParameter("domain_hint", "ucdavis.edu");
+
+                        return Task.FromResult(0);
+                    };
+                    options.Events.OnTokenValidated = async context =>
+                    {
+                        var identity = context.Principal.Identity as ClaimsIdentity;
+
+                        if (identity == null)
+                        {
+                            return;
+                        }
+
+                        // email comes across in both name claim and upn
+                        var email = identity.FindFirst(ClaimTypes.Upn).Value;
+
+                        // look up user info and add as claims
+                        var user = await _directorySearchService.GetByEmail(email);
```

Comment:
@jpknoll would like to get this service here instead of going the field route, but not sure how I can do it through the DI service

Reply:
Yeah, don't do this here. Do it on the AccountController.ExternalLoginCallback.